R/shiny.R

```diff
@@ -108,6 +110,13 @@ run <- function(file = "index.Rmd",
     }
   }
 
+  # If no `default_file` yet chosen but multiple .Rmd
+  # files exist in the directory, choose the first based
+  # on an alpha sort
+  if (is.null(default_file) && length(allRmds) > 1) {
```

Comment:
Sorry for not testing it earlier. I should have done it myself, but I just tried to reproduce #906, and it does not seem to be reproducible any more _without_ this PR (e.g. with the CRAN version), i.e., the issue seems to have been fixed somehow (note that it was filed two years ago). So I came back and looked at the first `if` statement above. I believe the bug in #906 should not exist any more because there was only one Rmd in that case:

```r
     if (length(allRmds) == 1) {
       # just one R Markdown document
       default_file <- allRmds
```

Therefore this PR was not fixing the bug, but introducing a new behavior instead.

I tend to revert this PR, but improve the documentation -- currently the last `else` clause is undocumented (i.e. the logic under the comment `# look for first one that has runtime: shiny`).